bindings/rust/s2n-tls/src/init.rs

```diff
@@ -93,7 +93,9 @@ mod mem {
             return s2n_status_code::FAILURE;
         };
 
-        dealloc(ptr as *mut _, layout);
+        if !ptr.is_null() {
```

Comment:
Do we need to still guard against it here? Could we replace this with a `debug_assert!(!ptr.is_null())`

Reply:
I don't think our bindings should assume we won't call free on a NULL pointer, because that's a completely normal thing to do in C.

If we don't want the duplicate check, I'd prefer to drop the one in the C. That one's arguably not _really_ correct.